JS/Lesson34/onur/my-app/src/components/NewRecipe/index.jsx

```diff
@@ -22,56 +23,195 @@ export const NewRecipe = () => {
 
 	const strCategory = watch('strCategory');
 
-	const onSubmit = (data) => console.log(data);
+
+
+
+	const onSubmit = (data) => {
+
+		if (data.strDrinkAlternate.trim() === '') {
+			data.strDrinkAlternate = null;
+		}
+		if (data.strImageSource.trim() === '') {
+			data.strImageSource = null;
+		}
+		if (data.strCreativeCommonsConfirmed.trim() === '') {
+			data.strCreativeCommonsConfirmed = null;
+		}
+		if (data.dateModified.trim() === '') {
+			data.dateModified = null;
+		}
+
+
+
+
+
+		const ingredientsArray = data.ingredients.split(",").map((ingredient) => ingredient.trim());
+		ingredientsArray.map((ingredient, index) => {
+			for (let i = 1; i <= 20; i++) {
+				if (ingredient.trim().length > 0 && (i <= ingredientsArray.length)) {
+					data[`strIngredient${index + 1}`] = ingredient;
+				} else {
+					data[`strIngredient${i}`] = "";
+				}
+			}
+		});
+		delete data.ingredients;
+
+		const measuresArray = data.measures.split(",").map((ingredient) => ingredient.trim());
+		measuresArray.map((measure, index) => {
+			for (let i = 1; i <= 20; i++) {
+				if (measure.trim().length > 0 && (i <= measuresArray.length)) {
+					data[`strMeasure${index + 1}`] = measure;
```

Comment:
Don't modify the incoming form data. Instead, create a new variable that will hold the form data that you later will send to the service or use in some other way in your application then add the computed values to that new variable.